src/random.cpp

```diff
@@ -585,6 +603,9 @@ static void ProcRand(unsigned char* out, int num, RNGLevel level)
     case RNGLevel::SLEEP:
         SeedSleep(hasher, rng);
         break;
+    case RNGLevel::PERIODIC:
```

Comment:
I don't get the point of this commit.
You're avoiding doing something that sleeps 1ms but runs in a tight loop for 10ms.
The real meat of the function gets called on a timer every minute, but also my be called at other times, but only runs if its been 10 minutes since the last run.

Maybe we should just skip the SeedInfrequent call in SeedSleep and then we can drop all the once-per-minute checking in this file wholesale and use the scheduler?

Reply:
I think you're right, there is fairly little advantage to having both RandAddSeedSleep and RandAddPeriodic. I've redone this PR to instead just have a RandAddPeriodic, which indeed simplifies a bunch of things including not keeping track of when the last run of the expensive stuff was (there is a still a counter in the perfmon data for windows, which only runs once every 10 minutes instead of every minute).